libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -2,12 +2,57 @@
 
 #include <eosio/chain/types.hpp>
 #include <eosio/chain/webassembly/common.hpp>
+#include <eosio/chain/webassembly/eos-vm.hpp>
+#include <eosio/chain/webassembly/preconditions.hpp>
+#ifdef EOSIO_EOS_VM_OC_RUNTIME_ENABLED
+#include <eosio/chain/webassembly/eos-vm-oc.hpp>
+#endif
 #include <fc/crypto/sha1.hpp>
 #include <boost/hana/string.hpp>
 
-namespace eosio { namespace chain {
-class apply_context;
-namespace webassembly {
+namespace eosio { namespace chain { namespace webassembly {
+
+   template <auto HostFunction, typename... Preconditions>
+   struct host_function_registrator {
+     template<typename Mod, typename Name>
+     constexpr host_function_registrator(Mod mod_name, Name fn_name) {
+         using rhf_t = eos_vm_host_functions_t;
+         rhf_t::add<HostFunction, Preconditions...>(mod_name.c_str(), fn_name.c_str());
+#ifdef EOSIO_EOS_VM_OC_RUNTIME_ENABLED
+         eosvmoc::register_eosvm_oc<HostFunction, std::tuple<Preconditions...>>(mod_name + BOOST_HANA_STRING(".") + fn_name);
+#endif
+      }
+   };
+
+#define REGISTER_HOST_FUNCTION(NAME, ...) \
```

Comment:
You should take the version that puts the host function registration in eos-vm.cpp.  This change was made to reduce compile times.

Reply:
Thanks. Moved.